mirrord/layer/src/debugger_ports.rs

```diff
@@ -205,41 +315,68 @@ impl DebuggerPorts {
+                env::set_var(MIRRORD_DETECT_DEBUGGER_PORT_ENV, next_type.to_string());
+            } else {
+                env::remove_var(MIRRORD_DETECT_DEBUGGER_PORT_ENV);
+            }
+            return Self::Detected(detected);
         }
 
-        let fixed_range = env::var(MIRRORD_IGNORE_DEBUGGER_PORTS_ENV)
+        // IGNORE_DEBUGGER_PORTS may have a combination of single, multiple or ranges of ports
+        // separated by a comma they need to be parsed individually
+        let mut v = vec![];
+        let _ : Option<()> = env::var(MIRRORD_IGNORE_DEBUGGER_PORTS_ENV)
             .ok()
             .and_then(|s| {
-                let chunks = s
-                    .split('-')
-                    .map(u16::from_str)
-                    .collect::<Result<Vec<_>, _>>()
-                    .inspect_err(|e| error!(
-                        "Failed to decode debugger ports from {} env variable: {}",
-                        MIRRORD_IGNORE_DEBUGGER_PORTS_ENV,
-                        e
-                    ))
-                    .ok()?;
-                match *chunks.as_slice() {
-                    [p] => Some(p..=p),
-                    [p1, p2] if p1 <= p2 => Some(p1..=p2),
-                    _ => {
-                        error!(
-                            "Failed to decode debugger ports from {} env variable: expected a port or a range of ports",
-                            MIRRORD_IGNORE_DEBUGGER_PORTS_ENV,
-                        );
-                        None
-                    },
-                }
+                s.split(',')
+                    .for_each(|s| {
+                        let chunks = s
+                            .split('-')
+                            .map(u16::from_str)
+                            .collect::<Result<Vec<_>, _>>()
+                            .inspect_err(|e| error!(
+                                "Failed to decode debugger port range from {} env variable: {}",
+                                MIRRORD_IGNORE_DEBUGGER_PORTS_ENV,
+                                e
+                            ))
+                            .ok().unwrap_or_default();
+                        let range = match *chunks.as_slice() {
+                            [p] => Some(p..=p),
+                            [p1, p2] if p1 <= p2 => Some(p1..=p2),
+                            _ => {
+                                error!(
+                                    "Failed to decode debugger ports from {} env variable: expected a port, list of ports or range of ports",
+                                    MIRRORD_IGNORE_DEBUGGER_PORTS_ENV,
+                                );
+                                None
+                            },
+                        };
+                        if let Some(ports) = range {
+                            v.push(Self::FixedRange(ports));
+                        }
+                    });
+                None
```

Comment:
I think that at this point we should implement `FromStr` and `fmt::Display` on `DebuggerPorts` :eyes:

Reply:
agreed, this code is gross